src/telliot_feeds/cli/utils.py

```diff
@@ -39,19 +68,35 @@ def valid_diva_chain(chain_id: int) -> bool:
 def build_feed_from_input() -> Optional[DataFeed[Any]]:
     """
     Build a DataFeed from CLI input
+    Called when the --build-feed flag is used on the telliot-feeds CLI
+
+    This function takes input from the user for the QueryType
+    and its matching QueryParameters, the builds a DataFeed
+    object if the QueryType is supported and the QueryParameters
+    can be casted to their appropriate data types from an input string.
+
+    Returns: DataFeed[Any]
+
     """
     try:
-        query_type = input("Enter a valid Query Type: ")
-        feed = DATAFEED_BUILDER_MAPPING[query_type]
+        msg = (
+            "Enter a valid QueryType from the options listed below:"
+            + "\n"
+            + "\n".join([i for i in DATAFEED_BUILDER_MAPPING.keys()])
+            + "\n"
+        )
+        query_type = input(msg)
+        feed: DataFeed[Any] = DATAFEED_BUILDER_MAPPING[query_type]
     except KeyError:
-        click.echo(f"No corresponding datafeed found for Query Type: {query_type}\n")
+        click.echo(f"No corresponding datafeed found for QueryType: {query_type}\n")
         return None
-    try:
-        for query_param in feed.query.__dict__.keys():
-            # accessing the datatype
-            param_dtype = feed.query.__annotations__[query_param]
-            val = input(f"Enter value for Query Parameter {query_param}: ")
+    for query_param in feed.query.__dict__.keys():
+        # accessing the datatype
+        type_hints = get_type_hints(feed.query)
+        param_dtype = get_args(type_hints[query_param])[0]  # parse out Optional type
+        val = input(f"Enter value for QueryParameter {query_param}: ")
 
+        try:
             if val is not None:
                 # cast input from string to datatype of query parameter
                 val = param_dtype(val)
```

Comment:
I mean it should literally just be around where you try to cast `val`, so:
```python
try: val = param_dtype(val)
except ValueError:
    click.echo("blah")
    return None
```

Reply:
ah gotcha